packages/web3-eth-contract/test/integration/contract_events.test.ts

```diff
@@ -0,0 +1,114 @@
+import { Web3BaseProvider } from 'web3-common';
+import { Contract } from '../../src';
+import { accounts } from '../shared_fixtures/integration_test_accounts';
+import { basicContractAbi, basicContractByteCode } from '../shared_fixtures/sources/Basic';
+import { processAsync } from '../shared_fixtures/utils';
+
+describe('contract', () => {
+	let contract: Contract<typeof basicContractAbi>;
+	let deployOptions: Record<string, unknown>;
+	let sendOptions: Record<string, unknown>;
+
+	beforeEach(async () => {
+		contract = new Contract(basicContractAbi, undefined, {
+			provider: 'ws://localhost:8545',
+		});
+
+		deployOptions = {
+			data: basicContractByteCode,
+			arguments: [10, 'string init value'],
+		};
+
+		sendOptions = { from: accounts[0].address, gas: '1000000' };
+
+		contract = await contract.deploy(deployOptions).send(sendOptions);
+	});
+
+	afterAll(async () => {
+		(contract.provider as Web3BaseProvider).disconnect(1000, 'message');
+	});
+
+	describe('events', () => {
+		it('should trigger the "contract.events.<eventName>"', async () => {
+			return expect(
+				processAsync(async resolve => {
+					const event = await contract.events.MultiValueEvent();
+
+					event.on('data', resolve);
+
+					// trigger event
+					await contract.methods
+						.firesMultiValueEvent('value', 12, true)
+						.send(sendOptions);
+				}),
+			).resolves.toEqual(
+				expect.objectContaining({
+					event: 'MultiValueEvent',
+				}),
+			);
+		});
+
+		it('should trigger the "contract.events.<eventName>" for indexed parameters', async () => {
+			return expect(
+				processAsync(async resolve => {
+					const event = await contract.events.MultiValueIndexedEvent({
+						filter: { val: 100 },
+					});
+
+					event.on('data', resolve);
+
+					// trigger event
+					await contract.methods
+						.firesMultiValueIndexedEvent('value', 12, true)
+						.send(sendOptions);
+					await contract.methods
+						.firesMultiValueIndexedEvent('value', 100, true)
+						.send(sendOptions);
+				}),
+			).resolves.toEqual(
+				expect.objectContaining({
+					event: 'MultiValueIndexedEvent',
+					returnValue: expect.objectContaining({ val: '100' }),
+				}),
+			);
+		});
+
+		it('should trigger when "fromBlock" is passed to contract.events.<eventName>', async () => {
+			return expect(
+				processAsync(async resolve => {
+					const event = await contract.events.MultiValueEvent({ fromBlock: 'latest' });
+
+					event.on('data', resolve);
+
+					// trigger event
+					await contract.methods
+						.firesMultiValueEvent('Event Value', 11, false)
+						.send(sendOptions);
+				}),
+			).resolves.toEqual(
+				expect.objectContaining({
```

Comment:
Does this work on browser tests? I think `objectContaining` is not exposed in expect in jasmine.

Reply:
`objectContaining` is part of the the default API for jest. It should work on all platforms. Will check if some issue occurred.

Reply:
Yes this is causing problem. Created a separate issue https://github.com/ChainSafe/web3.js/issues/4978 to address this to avoid more merge conflicts in this PR.